pkgs/tools/security/gnupg/21.nix

```diff
@@ -29,7 +29,8 @@ stdenv.mkDerivation rec {
     sed -i 's,"libpcsclite\.so[^"]*","${pcsclite}/lib/libpcsclite.so",g' scd/scdaemon.c
   ''; #" fix Emacs syntax highlighting :-(
 
-  configureFlags = optional x11Support "--with-pinentry-pgm=${pinentry}/bin/pinentry";
+  gpgPinentryBinaryPath = pinentry.gpgPinentryBinaryPath or "bin/pinentry";
+  configureFlags = optional (pinentry != null) "--with-pinentry-pgm=${pinentry}/${gpgPinentryBinaryPath}";
```

Comment:
Well, I suppose if someone does not want to have a dependency on pinentry, they should override with `pinentry = null`. To be honest, I’m not sure, what exactly this `x11Support` flag was used for, so I left it after renaming mostly for backward compatibility.

Actually, if that’s confusing, I can remove it completely. Unless someone can explain to me, why it was needed in the first place.